lib/index.js

```diff
@@ -28,124 +28,237 @@ var _typeMap = require('./typeMap');
+	var schema = schemaFromEndpointsEx(endpoints);
+	var rootType = new _graphql.GraphQLObjectType({
+		name: 'Query',
+		fields: function fields() {
+			return {
+				viewer: {
+					type: new _graphql.GraphQLObjectType({
+						name: 'viewer',
+						fields: schema.query
+					}),
+					resolve: function resolve() {
+						return 'Without this resolver graphql does not resolve further';
+					}
+				}
+			};
+		}
+	});
+
+	var graphQLSchema = {
+		query: rootType
+	};
+
+	if ((0, _keys2.default)(schema.mutation).length) {
+		graphQLSchema.mutation = new _graphql.GraphQLObjectType({
+			name: 'Mutation',
+			fields: schema.mutation
+		});
+	}
+
+	return new _graphql.GraphQLSchema(graphQLSchema);
 };
 
+var schemaFromEndpointsEx = function schemaFromEndpointsEx(endpoints) {
+	return {
+		query: function query() {
+			var queryFields = getQueriesFields(endpoints, false);
+			if (!(0, _keys2.default)(queryFields).length) {
+				throw new Error('Did not find any GET endpoints');
+			}
+			return queryFields;
+		},
+		mutation: getQueriesFields(endpoints, true)
+	};
+};
 
 var resolver = function resolver(endpoint) {
-  return function () {
-    var _ref = (0, _asyncToGenerator3.default)(_regenerator2.default.mark(function _callee(_, args, opts) {
-      var req, res;
-      return _regenerator2.default.wrap(function _callee$(_context) {
-        while (1) {
-          switch (_context.prev = _context.next) {
-            case 0:
-              req = endpoint.request(args, opts.GQLProxyBaseUrl);
-
-              if (opts.BearerToken) {
-                req.headers.Authorization = opts.BearerToken;
-              }
-              _context.next = 4;
-              return (0, _requestPromise2.default)(req);
-
-            case 4:
-              res = _context.sent;
-              return _context.abrupt('return', JSON.parse(res));
-
-            case 6:
-            case 'end':
-              return _context.stop();
-          }
-        }
-      }, _callee, undefined);
-    }));
-
-    return function (_x, _x2, _x3) {
-      return _ref.apply(this, arguments);
-    };
-  }();
+	return function () {
+		var _ref = (0, _asyncToGenerator3.default)( /*#__PURE__*/_regenerator2.default.mark(function _callee(_, args, opts) {
+			var req, res;
+			return _regenerator2.default.wrap(function _callee$(_context) {
+				while (1) {
+					switch (_context.prev = _context.next) {
+						case 0:
+							req = endpoint.request(args, opts.GQLProxyBaseUrl);
+
+							if (opts.BearerToken) {
+								req.headers.Authorization = opts.BearerToken;
+							}
+							_context.next = 4;
+							return (0, _requestPromise2.default)(req);
+
+						case 4:
+							res = _context.sent;
+							return _context.abrupt('return', JSON.parse(res));
+
+						case 6:
+						case 'end':
+							return _context.stop();
+					}
+				}
+			}, _callee, undefined);
+		}));
+
+		return function (_x, _x2, _x3) {
+			return _ref.apply(this, arguments);
+		};
+	}();
 };
```

Comment:
I'm assuming you're running `prettier` or some formatter? It makes this PR very noise, and hard to see the **actual** functional changes.

Reply:
I realize that now ...

Reply:
@mihaislobozeanu could you do a fix for that?

Reply:
I will do that next week, but first I need the fix for node-request-by-swagger, I have created a pull request for that.

Thanks,
Mihai

Reply:
Could you please explain what exactly requires changes in node-request-by-swagger? Thanks

Reply:
@yarax Check his [PR](https://github.com/yarax/node-request-by-swagger/pull/2) on your other repo for node-request-by-swagger.

> It seems that the package request does not accept js Object in body, it accepts stream or string. I fixed it.

Reply:
Yes, that's the one

Reply:
@mihaislobozeanu but what if the value is not stringifiable? I would say that the value should be prepared before passing to `node-request-by-swagger`, because specification doesn't say that if param.in is `body` then it must be a json string

Reply:
Sorry for the delayed answer. As I said in my pull request comment, you are using 'request' library for the http request. It does not accept requests with the body as Object. It only accepts streams or strings.
I don't know how to answer your question, I've never encountered such situation. As you can see, I apply stringify only if (contentType === 'application/x-www-form-urlencoded').

Reply:
Ok, I've rollbacked my change. There is no problem when query but on mutation. I have a node server with express. among which I have a bodyParser middleware. Therefore the body is the object type. If you go through the code of the request library, on the request.js file, at line 416 there is setContentLength method. The body is not a string, is not an array, not a stream and it does not have the length property. For those reasons, you'll get an error.
Explaining this to you, I have understood your point. I will change and only if the content type is application/json I will stringify the body.